parse.y

```diff
@@ -2587,7 +2587,81 @@ rb_parser_string_hash_cmp(rb_parser_string_t *str1, rb_parser_string_t *str2)
             memcmp(ptr1, ptr2, len1) != 0);
 }
 #endif
-#endif
+
+static void
+rb_parser_ary_extend(rb_parser_t *p, rb_parser_ary_t *ary, long len)
+{
+    long i;
+    if (ary->capa < len) {
+        ary->capa = len;
+        if (ary->data == NULL) {
+            ary->data = xcalloc(len, sizeof(void *));
+        } else {
+            ary->data = xrealloc(ary->data, sizeof(void *) * len);
+        }
```

Comment:
`xrealloc` allocates a new memory when the pointer is NULL, as well as the standard `realloc`.

```suggestion
        ary->data = xrealloc(ary->data, sizeof(void *) * len);
```